components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -318,3 +326,28 @@ func toDbOIDCSpec(oauth2Config *v1.OAuth2Config, oidcConfig *v1.OIDCConfig) db.O
 		Scopes:       append([]string{goidc.ScopeOpenID, "profile", "email"}, oauth2Config.GetScopes()...),
 	}
 }
+
+func assertIssuerIsReachable(host string) error {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		Proxy:           http.ProxyFromEnvironment,
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   2 * time.Second,
+		// never follow redirects
+		CheckRedirect: func(*http.Request, []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	resp, err := client.Get(host)
+	if err != nil {
+		return err
+	}
+	resp.Body.Close()
+	if resp.StatusCode > 499 {
+		return fmt.Errorf("returned status %d", resp.StatusCode)
```

Comment:
```suggestion
		return fmt.Errorf("OIDC reachability check returned status %d", resp.StatusCode)
```

Reply:
This ends up being the public facing error message, so we should communicate as much detail as possible.